### ISSUE TITLE
gist sources are unavailable

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#2460]
Looks like the notebook tree refactoring disconnected some logic to do with with gist sources (foreign notebooks).

`&source=foo` should allow loading notebooks from other gist backends, as specified in rcloud.conf. It looks like the initialization has moved to the notebook tree model, but editor_tab is still looking for `gist_sources_` locally, so it dies.

(Noticed this because I was going to take a quick look at #2444.)

**Comment by gordonwoodhull** [comment]
You can test a trivial configuration of foreign notebooks / gist sources with an `rcloud.conf` like this:

```
host: 127.0.0.1
github.client.id: XXX
github.client.secret: XXX
github.base.url: https://github.com/
github.api.url: https://api.github.com/
github.gist.url: https://gist.github.com/
rcs.engine: redis
rcs.redis.host: 127.0.0.1:6379
rcs.system.config.addons: rcloud.viewer, rcloud.enviewer, rcloud.notebook.info
rcloud.languages: rcloud.r, rcloud.python, rcloud.rmarkdown
rcloud.deployment: Here
rcloud.deployment.color: blue

gist.source: hubgit
github.client.id: XXX
github.client.secret: XXX
github.base.url: https://github.com/
github.api.url: https://api.github.com/
github.gist.url: https://gist.github.com/
```

Note the line break: that's what delimits sections in DCF. I'm not sure if the client id/secret has to be different; for some reason I've done so in my config.

Once you have that in place, try loading a notebook which is not in your current tree with

```
?notebook=XXX&source=hubgit
```

You should see it appear within the regular tree but in green.

**Comment by shaneporter** [comment]
Thanks @gordonwoodhull - I'll give that a try.

**Comment by shaneporter** [comment]
I tried doing this by creating another user in github, setting up the RCloud auth, and creating a notebook.

I created the `gist.source` section, ensuring that the other user's client id and secret were there. 

Following some code changes, I don't see an error, but the notebook appears under the All Notebooks/otheruser's notebooks part of the tree.

I also tried duplicating the section as you suggested, but I'm not sure how to load a notebook which is not already in the tree if the notebooks are pointing to the same user.

Sorry if I've missed something obvious here.
